lib/game/board.rb

```diff
@@ -0,0 +1,55 @@
+module Game
+  class Board
+    EMPTY_BOARD = (0..8).to_a
+    PLAYER_ONE = 'X'
+    PLAYER_TWO = 'O'
+
+    attr_reader :error, :positions
+
+    def initialize(positions = EMPTY_BOARD)
+      @error = nil
+      @positions = positions
+    end
+
+    def available_positions
+      @positions.select { |pos| pos.is_a?(Integer) }
+    end
+
+    def place_token(position, player)
+      position_valid?(position) ? apply_token(position, player) : apply_errors(position)
+    end
+
+    def has_error?
+      @error != nil
+    end
+
+    private
+
+    def position_valid?(position)
+      is_integer?(position) && available_positions.include?(position.to_i)
+    end
+
+    def validation_reason(position)
+      !is_integer?(position) || is_outside_board_range?(position) ? :invalid_position : :position_taken
+    end
+
+    def is_integer?(position)
```

Comment:
In ruby, usually we don't have the `is_` prefix as the question mark works

Reply:
Thanks, I'll bear that in mind.